python/model_hosting_container_standards/sagemaker/__init__.py

```
@@ -79,11 +93,23 @@ def inject_adapter_id(adapter_path: str):
     if not isinstance(adapter_path, str):
         logger.exception("adapter_path must be a string")
         raise ValueError("adapter_path must be a string")
-    # create request_shape
-    request_shape = {}
-    request_shape[adapter_path] = (
-        f'headers."{SageMakerLoRAApiHeader.ADAPTER_IDENTIFIER}"'
-    )
+    if append and separator is None:
```

Let's fail fast if separator is specified without append